controllers/result.go

```diff
@@ -0,0 +1,44 @@
+package controllers
+
+import (
+	"net/http"
+
+	"go-google-scraper-challenge/services/scraper"
+
+	"github.com/beego/beego/v2/server/web"
+)
+
+// ResultController operations for User
+type ResultController struct {
+	BaseController
+}
+
+// URLMapping map user controller actions to functions
+func (c *ResultController) URLMapping() {
+	c.Mapping("List", c.List)
+	c.Mapping("Create", c.Create)
+}
+
+func (c *ResultController) List() {
+	c.EnsureAuthenticatedUser(true)
+
+	c.Layout = "layouts/default.html"
+	c.TplName = "search_results/list.html"
+
+	web.ReadFromRequest(&c.Controller)
+}
+
+func (c *ResultController) Create() {
+	c.EnsureAuthenticatedUser(false)
```

Comment:
Should not an authenticated user require to perform search/upload? Then associate data/result with the user and show these data in the user own dashboard?

Reply:
@junan Actually passing the `false` doesn't mean to not check for user authentication but it's the `redirect` param to tell whether to redirect or not if the user is not authenticated

```
func (base *BaseController) EnsureAuthenticatedUser(redirect bool) {
	currentUser := base.GetCurrentUser()
	if currentUser == nil {
		if redirect {
			base.Controller.Redirect("/signin", http.StatusFound)
		} else {
			base.Controller.Abort(fmt.Sprint(http.StatusMethodNotAllowed))
		}
	}
	base.Controller.Data["CurrentUser"] = currentUser
}
```

But I know this is misleading, at first I also confuse myself 😅  so how about renaming the function instead? but it would end up having 2 functions `EnsureAuthenticatedUserWithRedirect` and `EnsureAuthenticatedUser`
What do you think?

Reply:
@carryall To me, renaming is better because `c.EnsureAuthenticatedUser(false)` is confusing 😄

Reply:
@carryall Yes, checking the action name will be even better. I also [did](https://github.com/junan/google-scraper/pull/27/files#diff-cda711af0e0738fc95ed8c53339e1b4dc77eb4e956c0eedb7cc5bc1be3b35e9dR64) the similar action name setting and checking (mine was for checking authentication require or not)

Reply:
✅  done on [627baac](https://github.com/carryall/go-google-scraper-challenge/pull/34/commits/627baac114053b21f4ad8315a93bea02d1a92f42)